numpy/fft/helper.py

```diff
@@ -253,26 +254,29 @@ def __init__(self, max_size_in_mb, max_item_count):
         # Much simpler than inheriting from it and having to work around
         # recursive behaviour.
         self._dict = OrderedDict()
+        self._lock = threading.Lock()
 
     def setdefault(self, key, value):
         return self._dict.setdefault(key, value)
 
     def __getitem__(self, key):
-        # pop + add to move it to the end.
-        value = self._dict.pop(key)
-        self._dict[key] = value
-        self._prune_dict()
-        return value
+        with self._lock:
+            # pop + add to move it to the end.
+            value = self._dict.pop(key)
```

Comment:
It will just raise a `KeyError` which is the same behavior as calling `__getitem__()` of a dictionary with a non-existing key.

Lots of methods are missing but at least `__getitem__()` and `__setitem__()` mimic the behavior of a normal `dict` making the new cache a drop-in replacement for the old simple dictionary cache.

The case of a missing key is handled in the `_raw_fft()` method in `fftpack.py`.